### PR TITLE
Changed submodule and added env variables for AWS

### LINKED ISSUE #10: Update portal to new submodule location and add AWS credential related env variables
Refer to `README.md` for instructions

things i dont like about doing it this way:
- windows users take a massive L because the volume mounting for `~/.aws` is different on windows
- requires devs to install aws-cli on their local machines which goes away from our goal of zero setup outside of docker
- there's no way for us to enforce aws profile names so users that are setting up using `aws configure sso` MUST follow the commands to the tea or their backend container will just flat out fail.

there really doesn't seem to be a better way of doing this based on my research unless we do some really jank stuff with iam users and roles.

my end goal is following this IF we end up using AWS ECS: https://aws.amazon.com/blogs/compute/a-guide-to-locally-testing-containers-with-amazon-ecs-local-endpoints-and-docker-compose/

(this is essentially the same thing but it's needed when we deploy to ECS so it can dynamically grab credentials if you're working locally or running on AWS)

### PR DESCRIPTION
**Pull request by KevinHa48**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,41 @@
...
+
+AA-DR Portal requires AWS-CLI in order to get credentials needed to make AWS service related calls.
+
+1. [Install AWS-CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) for your respective OS (using Windows will require a configuration change)
+2. (**For Windows Users**) Go to `docker-compose.yml` and change this line in the backend container configuration:
+
+```
+    volumes:
+      - ./backend:/app
+      - ~/.aws:/root/.aws <- change to %UserProfile%\.aws:/root/.aws
```
Comment (gcarvellas) [inline README.md]: Does `~/.aws` not work in windows powershell?

**Comment by gcarvellas** [comment]
doesn't work

Version:
```
aws-cli/2.13.21 Python/3.11.5 Linux/6.5.4-arch2-1 source/x86_64.arch prompt/off
6.5.4-arch2-1 
```

@g00gol can you test this too? Follow the readme for instructions.

Try running the code. I get this error:
```
botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [11] [INFO] Worker exiting (pid: 11)
aa-dr-backend   | ERROR:root:Couldn't list users for us-east-1_VR47Oj8cS. Here's why: ExpiredTokenException: The security token included in the request is expired
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [ERROR] Exception in worker process
aa-dr-backend   | Traceback (most recent call last):
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
aa-dr-backend   |     worker.init_process()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/gthread.py", line 95, in init_process
aa-dr-backend   |     super().init_process()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
aa-dr-backend   |     self.load_wsgi()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
aa-dr-backend   |     self.wsgi = self.app.wsgi()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
aa-dr-backend   |     self.callable = self.load()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
aa-dr-backend   |     return self.load_wsgiapp()
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
aa-dr-backend   |     return util.import_app(self.app_uri)
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/util.py", line 371, in import_app
aa-dr-backend   |     mod = importlib.import_module(module)
aa-dr-backend   |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
aa-dr-backend   |     return _bootstrap._gcd_import(name[level:], package, level)
aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
aa-dr-backend   |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
aa-dr-backend   |   File "/app/app.py", line 19, in <module>
aa-dr-backend   |     print(CognitoIdentityProviderWrapper().get_user("test"))
aa-dr-backend   |   File "/app/database/cognito.py", line 20, in get_user
aa-dr-backend   |     response = self.cognito_idp_client.admin_get_user(UserPoolId=COGNITO_USERPOOL_ID, Username=username)
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 535, in _api_call
aa-dr-backend   |     return self._make_api_call(operation_name, kwargs)
aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 980, in _make_api_call
aa-dr-backend   |     raise error_class(parsed_response, operation_name)
aa-dr-backend   | botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [INFO] Worker exiting (pid: 12)
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:11) exited with code 3
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:12) was sent SIGTERM!
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Shutting down: Master
aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Reason: Worker failed to boot.
aa-dr-backend   | make: *** [Makefile:9: run] Error 3
aa-dr-backend exited with code 2
```

**Comment by gcarvellas** [comment]
> doesn't work
> 
> Version:
> 
> ```
> aws-cli/2.13.21 Python/3.11.5 Linux/6.5.4-arch2-1 source/x86_64.arch prompt/off
> 6.5.4-arch2-1 
> ```
> 
> @g00gol can you test this too? Follow the readme for instructions.
> 
> Try running the code. I get this error:
> 
> ```
> botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [11] [INFO] Worker exiting (pid: 11)
> aa-dr-backend   | ERROR:root:Couldn't list users for us-east-1_VR47Oj8cS. Here's why: ExpiredTokenException: The security token included in the request is expired
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [ERROR] Exception in worker process
> aa-dr-backend   | Traceback (most recent call last):
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
> aa-dr-backend   |     worker.init_process()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/gthread.py", line 95, in init_process
> aa-dr-backend   |     super().init_process()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
> aa-dr-backend   |     self.load_wsgi()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
> aa-dr-backend   |     self.wsgi = self.app.wsgi()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
> aa-dr-backend   |     self.callable = self.load()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
> aa-dr-backend   |     return self.load_wsgiapp()
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
> aa-dr-backend   |     return util.import_app(self.app_uri)
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/util.py", line 371, in import_app
> aa-dr-backend   |     mod = importlib.import_module(module)
> aa-dr-backend   |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
> aa-dr-backend   |     return _bootstrap._gcd_import(name[level:], package, level)
> aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
> aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
> aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
> aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
> aa-dr-backend   |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
> aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
> aa-dr-backend   |   File "/app/app.py", line 19, in <module>
> aa-dr-backend   |     print(CognitoIdentityProviderWrapper().get_user("test"))
> aa-dr-backend   |   File "/app/database/cognito.py", line 20, in get_user
> aa-dr-backend   |     response = self.cognito_idp_client.admin_get_user(UserPoolId=COGNITO_USERPOOL_ID, Username=username)
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 535, in _api_call
> aa-dr-backend   |     return self._make_api_call(operation_name, kwargs)
> aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 980, in _make_api_call
> aa-dr-backend   |     raise error_class(parsed_response, operation_name)
> aa-dr-backend   | botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [INFO] Worker exiting (pid: 12)
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:11) exited with code 3
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:12) was sent SIGTERM!
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Shutting down: Master
> aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Reason: Worker failed to boot.
> aa-dr-backend   | make: *** [Makefile:9: run] Error 3
> aa-dr-backend exited with code 2
> ```

nvm I'm a clown. Note for the future: environment variables take precedence over aws-cli

**Comment by KevinHa48** [comment]
> > doesn't work
> > Version:
> > ```
> > aws-cli/2.13.21 Python/3.11.5 Linux/6.5.4-arch2-1 source/x86_64.arch prompt/off
> > 6.5.4-arch2-1 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @g00gol can you test this too? Follow the readme for instructions.
> > Try running the code. I get this error:
> > ```
> > botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [11] [INFO] Worker exiting (pid: 11)
> > aa-dr-backend   | ERROR:root:Couldn't list users for us-east-1_VR47Oj8cS. Here's why: ExpiredTokenException: The security token included in the request is expired
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [ERROR] Exception in worker process
> > aa-dr-backend   | Traceback (most recent call last):
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
> > aa-dr-backend   |     worker.init_process()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/gthread.py", line 95, in init_process
> > aa-dr-backend   |     super().init_process()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
> > aa-dr-backend   |     self.load_wsgi()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
> > aa-dr-backend   |     self.wsgi = self.app.wsgi()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
> > aa-dr-backend   |     self.callable = self.load()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
> > aa-dr-backend   |     return self.load_wsgiapp()
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
> > aa-dr-backend   |     return util.import_app(self.app_uri)
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/gunicorn/util.py", line 371, in import_app
> > aa-dr-backend   |     mod = importlib.import_module(module)
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
> > aa-dr-backend   |     return _bootstrap._gcd_import(name[level:], package, level)
> > aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
> > aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
> > aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
> > aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
> > aa-dr-backend   |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
> > aa-dr-backend   |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
> > aa-dr-backend   |   File "/app/app.py", line 19, in <module>
> > aa-dr-backend   |     print(CognitoIdentityProviderWrapper().get_user("test"))
> > aa-dr-backend   |   File "/app/database/cognito.py", line 20, in get_user
> > aa-dr-backend   |     response = self.cognito_idp_client.admin_get_user(UserPoolId=COGNITO_USERPOOL_ID, Username=username)
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 535, in _api_call
> > aa-dr-backend   |     return self._make_api_call(operation_name, kwargs)
> > aa-dr-backend   |   File "/usr/local/lib/python3.10/site-packages/botocore/client.py", line 980, in _make_api_call
> > aa-dr-backend   |     raise error_class(parsed_response, operation_name)
> > aa-dr-backend   | botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the AdminGetUser operation: The security token included in the request is expired
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [12] [INFO] Worker exiting (pid: 12)
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:11) exited with code 3
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Worker (pid:12) was sent SIGTERM!
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Shutting down: Master
> > aa-dr-backend   | [2023-09-29 03:39:29 +0000] [10] [ERROR] Reason: Worker failed to boot.
> > aa-dr-backend   | make: *** [Makefile:9: run] Error 3
> > aa-dr-backend exited with code 2
> > ```
> 
> nvm I'm a clown. Note for the future: environment variables take precedence over aws-cli

important article for the future:

https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html#cli-chap-authentication-precedence

**Pull request merged**